.github/workflows/publish-to-pypi.yml

```
@@ -17,8 +17,14 @@ name: Publish to PyPI
 # Only run for pushes to the main branch and releases.
 on:
   push:
-    branches:
-      - main
+    branches: [ main ]
+    paths:
+      - 'pygmt/**'
```

Need to exclude `pygmt/tests/**` if we make the decision in #2956.

---

Add this after merging #2957

```suggestion
      - 'pygmt/**'
      - '!pygmt/tests/**'
```